src/featbin/paste-vectors.cc

```diff
@@ -0,0 +1,129 @@
+// featbin/paste-vectors.cc
+
+// Copyright 2020 Ivan Medennikov (STC-innovation Ltd)
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include "base/kaldi-common.h"
+#include "matrix/kaldi-matrix.h"
+#include "util/common-utils.h"
+
+namespace kaldi {
+
+void AppendVectors(const std::vector<Vector<BaseFloat> > &in,
+                   Vector<BaseFloat> *out) {
+  // Check the lengths
+  int32 tot_dim = in[0].Dim();
+  for (int32 i = 1; i < in.size(); ++i) {
+    int32 dim = in[i].Dim();
+    tot_dim += dim;
+  }
+  out->Resize(tot_dim);
+  int32 dim_offset = 0;
+  for (int32 i = 0; i < in.size(); ++i) {
+    int32 this_dim = in[i].Dim();
+    out->Range(dim_offset, this_dim).CopyFromVec(
+        in[i].Range(0, this_dim));
+    dim_offset += this_dim;
+  }
+}
+
+
+}  // namespace kaldi
+
+int main(int argc, char *argv[]) {
+  try {
+    using namespace kaldi;
+    using namespace std;
+
+    const char *usage =
+        "Concatenate vector files \n"
```

Comment:
Let's put it this way (feel free to break into lines 60-100 characters on output each paragraph, so they are nice and readable):

>Append to vectors the first archive the vectors under the same key from the second and following archives, and output the result.
>
>NOTE: Keys absent from the first archive will not be present in output, even if they occur in other input archives."

The space before '\n' is not needed.